File: python/KRAKEN/Tests/DW/Manual/elsed.m
clear all, close all

addpath /home/orodrig/FORdoc/at/Matlab/ReadWrite

disp('KRAKEN - Elastic sediment manual case')

system('cp  field.flp elsed.flp');
system('kraken.exe elsed');
%system('krakenc.exe elsed');
system('fields.exe elsed < elsed.flp');

[ PlotTitle, PlotType, freq, atten, Pos, p ] = read_shd( 'elsed.shd' );

zs = Pos.s.depth;

rarray = Pos.r.range; rarraykm = rarray/1000;
zarray = Pos.r.depth;

rmax = max( rarray ); rmaxkm = rmax/1000;
zmax = max( zarray );

pressure = squeeze( p );

tl = -20*log10( abs( pressure ) ); 

figure(1)
plot(rarraykm,tl)
xlabel('Range (km)')
ylabel('TL (dB)')
title('KRAKEN - Elastic sediment manual case')
grid on, box on
axis ij

disp('done.')

File: python/KRAKEN/Tests/DW/Manual/elsed.py
#==================================================================
#  
#  KRAKEN: Elastic sediment manual case
#  Mexilhoeira Grande, Sab Jul 29 10:19:09 WEST 2017
#  Written by Tordar
#  
#==================================================================

# ipython: run elsed

from os import *
import sys
from numpy import *
from scipy.io import *
from pylab import *
sys.path.append ("/home/orodrig/FORdoc/at/Python/")
from readshd import *

rs = 0.0

print("KRAKEN - Elastic sediment manual case")

system("kraken.exe elsed")
system("fields.exe elsed < elsed.flp")

filename = 'elsed.shd'
xs = nan
ys = nan
pressure,geometry = readshd(filename,xs,ys)

zs     = geometry["zs"]
rarray = geometry["rarray"]; rarraykm = rarray/1000
zarray = geometry["zarray"]

Dmax = zarray[-1]
rmax = rarray[-1]; rmaxkm = rmax/1000

p = squeeze( pressure )
tl = -20*log10( abs( p ) )
tlmin = min( tl )
tlmax = max( tl )

figure(1)
plot(rarraykm,tl)
xlabel('Range (km)')
ylabel('TL (dB)')
title('KRAKEN - Elastic sediment manual case')
ylim(tlmax,tlmin)
grid(True)

show()

print("done.")
